todo/views/role.py

```diff
@@ -0,0 +1,239 @@
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from rest_framework import status
+from rest_framework.request import Request
+from django.conf import settings
+from drf_spectacular.utils import extend_schema, OpenApiParameter, OpenApiResponse
+from drf_spectacular.types import OpenApiTypes
+
+from todo.serializers.create_role_serializer import CreateRoleSerializer
+from todo.serializers.update_role_serializer import UpdateRoleSerializer
+from todo.serializers.get_roles_serializer import RoleQuerySerializer
+from todo.services.role_service import RoleService
+from todo.exceptions.role_exceptions import RoleNotFoundException, RoleAlreadyExistsException
+
+
+class RoleListView(APIView):
+    @extend_schema(
+        operation_id="get_roles",
+        summary="Get all roles",
+        description="Retrieve all roles with optional filtering",
+        tags=["roles"],
+        parameters=[
+            OpenApiParameter(
+                name="is_active",
+                type=OpenApiTypes.BOOL,
+                location=OpenApiParameter.QUERY,
+                description="Filter by active status",
+            ),
+            OpenApiParameter(
+                name="type",
+                type=OpenApiTypes.STR,
+                location=OpenApiParameter.QUERY,
+                description="Filter by role type",
+            ),
+            OpenApiParameter(
+                name="scope",
+                type=OpenApiTypes.STR,
+                location=OpenApiParameter.QUERY,
+                description="Filter by role scope (GLOBAL/TEAM)",
+            ),
+        ],
+        responses={
+            200: OpenApiResponse(description="Roles retrieved successfully"),
+            400: OpenApiResponse(description="Bad request"),
+            500: OpenApiResponse(description="Internal server error"),
+        },
+    )
+    def get(self, request: Request):
+        try:
+            query_serializer = RoleQuerySerializer(data=request.query_params)
+            query_serializer.is_valid(raise_exception=True)
+
+            filters = {}
+            if query_serializer.validated_data.get("is_active") is not None:
+                filters["is_active"] = query_serializer.validated_data["is_active"]
+            if query_serializer.validated_data.get("type"):
+                filters["type"] = query_serializer.validated_data["type"]
+            if query_serializer.validated_data.get("scope"):
+                filters["scope"] = query_serializer.validated_data["scope"]
```

Comment:
- should we create a method for handling the filters and use it here to make this function look clean?

Reply:
done